TTPChallenge/QuizVC.swift

```diff
@@ -0,0 +1,249 @@
+                it[row] = Answer.yes
+            default:
+                break
+            }
+        case 2:
+            switch row {
+            case 0, 1, 2, 3:
+                wd[row] = Answer.yes
+            default:
+                break
+            }
+        case 3:
+            switch row {
+            case 0, 1, 2, 3:
+                md[row] = Answer.yes
+            default:
+                break
+            }
+        case 4:
+            switch row {
+            case 0, 1, 2, 3:
+                da[row] = Answer.yes
+            default:
+                break
+            }
+        default:
+            break
+        }
+        self.activateResultButton()
+    }
+    
+    func noForEachTechType(section: Int, row: Int){
+        
+        switch section {
+        case 0:
+            switch row {
+            case 0, 1, 2, 3:
+                qa[row] = Answer.no
+            default:
+                break
+            }
+        case 1:
+            switch row {
+            case 0, 1, 2, 3:
+                it[row] = Answer.no
+            default:
+                break
+            }
+        case 2:
+            switch row {
+            case 0, 1, 2, 3:
+                wd[row] = Answer.no
+            default:
+                break
+            }
+        case 3:
+            switch row {
+            case 0, 1, 2, 3:
+                md[row] = Answer.no
+            default:
+                break
+            }
+        case 4:
+            switch row {
+            case 0, 1, 2, 3:
+                da[row] = Answer.no
+            default:
+                break
+            }
+        default:
+            break
+        }
+
+        DispatchQueue.main.async {
+            self.activateResultButton()
+        }
+    }
+
+    func activateResultButton(){
+        let qaCount = qa.filter{$0 == Answer.none}.count
+        let itCount = it.filter{$0 == Answer.none}.count
+        let wdCount = wd.filter{$0 == Answer.none}.count
+        let mdCount = md.filter{$0 == Answer.none}.count
+        let daCount = da.filter{$0 == Answer.none}.count
+        
+        let allQuestionsToBeResponded = qa.count + it.count + wd.count + md.count + da.count
+        let allEmptyStrs = qaCount + itCount + wdCount + mdCount + daCount
+        respondedQuestions = 20 - allEmptyStrs
+    
+        if allEmptyStrs > 0 {
+            nextButton.isEnabled = false
+        } else {
+            nextButton.isEnabled = true
+        }
+        progressLabel.text = "\(respondedQuestions)/\(allQuestionsToBeResponded)"
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == Identifier.Segue.toResult {
+            let destVC = segue.destination as! QuizResultVC
+            DispatchQueue.main.async {
+                destVC.indexArrayPassed = self.jobIndexArray
+                destVC.techTypesFromResults = self.techJobsForResult
+            }
```

Comment:
Hmmm this guy wasn't updating unless it was handled with DispatchQue. There was a concurrency issue that segue was done before passing data to the destination VC. So I put them in the main thread that the destination VC ( resultVC) could be updated properly on the UI. Any thoughts?